lib/stores/list_store.dart

```diff
@@ -48,6 +67,26 @@ class ListProvider with ChangeNotifier {
     });
   }
 
+  Future<void> loadBuckets({BuildContext context, int listId, int page = 1}) {
+    _buckets = [];
+    _isLoading = true;
+    notifyListeners();
+
+    Map<String, List<String>> queryParams = {
+      "page": [page.toString()]
+    };
+
+    return VikunjaGlobal.of(context).bucketService.getAllByList(listId, queryParams).then((response) {
+      if (response.headers["x-pagination-total-pages"] != null) {
+        _maxPages = int.parse(response.headers["x-pagination-total-pages"]);
```

Comment:
Shouldn't this be handled in the service?

Reply:
Probably, honestly when I wrote this I was just copying from `loadTasks` above. Now that I'm looking at it again though, I just realized it clears the existing buckets when it goes to load the next page which is not good. @Benimautner and I were talking about changing the state management stuff anyway, but that will have to be a future PR.

Reply:
I don't know why pagination is handled there rather than in the service. As @k9withabone said, we should restructure the state management and handle pagination in the service.
I will open an issue regarding state management soon so we can discuss possible options for the rework.